File: Leetcode_Python/decode-string.py
class Solution(object):
    def decodeString(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []
        curr_str = ""
        num = 0

        for char in s:
            if char.isdigit():
                num = num * 10 + int(char)  # Handle multi-digit numbers
            elif char == "[":
                stack.append(curr_str)
                stack.append(num)
                curr_str = ""
                num = 0
            elif char == "]":
                repeat_count = stack.pop()
                prev_str = stack.pop()
                curr_str = prev_str + (curr_str * repeat_count)
            else:
                curr_str += char

        return curr_str

File: Leetcode_TS/decode-string.ts
function decodeStrin(s: string): string {
    
    let stack: (string | number)[] = []
    let str: string = ""
    let num: number = 0

    for (let i = 0; i < s.length; i++) {
            if (!isNaN(Number(s[i]))) {
                num = num * 10 + Number(s[i])
            } else if (s[i] === "[") {
                stack.push(str)
                stack.push(num)
                str = ""
                num = 0
            } else if (s[i] === "]") {
                const num = stack.pop() as number
                const prevStr = stack.pop() as string
                str = prevStr + str.repeat(num)
            } else {
                str += s[i]
            }
    }
    return str
};